third_party/proton/dialect/lib/ProtonGPUToLLVM/ProtonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -12,4 +15,30 @@ Value TargetInfo::clock(ConversionPatternRewriter &rewriter, Location loc,
   return Value();
 }
 
+Value TargetInfo::hardwareId(ConversionPatternRewriter &rewriter,
```

Comment:
`hardwareId` is a bit confusing name, `processorId` might be better

Reply:
This is the HW Reg ID that is used to hold various CU related information, so probable hardwareRegId is make more sense:
https://llvm.org/docs/AMDGPU/gfx9_hwreg.html

Reply:
Yeah, I think the return value is composed of multiple "hardware ids" in the AMD backend.
But the NVIDIA counterpart only returns `sm_id`.
If `sm_id` is all we need, we could shift the return value and get `sm_id` as well for the AMD backend?

Reply:
Yeah, cu_id is just:
```
int cu_id = (hw_id & 0xf00) >> 8;
```
So in the pass just replace with something like:
```
auto hwreg = builder.newConstantOperand("hwreg(HW_REG_HW_ID, 0, 32)");
auto masked = builder.and_(hwreg, builder.i32_val(0xf00));
auto cu_id = build.lshr(masked, builder.i32_val(8));
```